src/main/scala/org/broadinstitute/dsde/firecloud/core/DeleteEntities.scala

```diff
@@ -21,41 +23,44 @@ object DeleteEntities {
     Props(new DeleteEntitiesActor(requestContext, entities))
 }
 
-class DeleteEntitiesActor(requestContext: RequestContext, entities: Seq[EntityId]) extends Actor with Aggregator with FireCloudRequestBuilding {
+class DeleteEntitiesActor(requestContext: RequestContext, entities: Seq[EntityId]) extends Actor with FireCloudRequestBuilding {
 
   implicit val system = context.system
   import system.dispatcher
   val log = Logging(system, getClass)
 
-  expectOnce {
+  def receive = {
     case ProcessUrl(baseUrl: String) =>
       log.debug(s"Deleting ${entities.size} entities for url: " + baseUrl)
       val pipeline = authHeaders(requestContext) ~> sendReceive
       val deleteUrls = entities.map(e => FireCloudDirectiveUtils.encodeUri(s"$baseUrl/${e.entityType}/${e.entityName}"))
       val deleteFutures: Seq[Future[HttpResponse]] = deleteUrls.map(url => pipeline { Delete(url) })
-      Future sequence deleteFutures onComplete {
-        case Success(responses) if responses.forall(_.status.isSuccess) =>
-          context.parent ! RequestComplete(OK)
-          context stop self
-        case Success(responses) =>
-          val errors = responses.filterNot(_.status.isSuccess) map { e => (e, ErrorReport.tryUnmarshal(e)) }
-          val errorReports = errors collect { case (_, Success(report)) => report }
-          val missingReports = errors collect { case (originalError, Failure(_)) => originalError }
-
-          val errorMessage = {
-            val baseMessage = "%d entities deleted, with %d failures.  Errors: %s".format(entities.size, errors.size, errors mkString ",")
-            if (missingReports.isEmpty) baseMessage
-            else {
-              val supplementalErrorMessage = "Additionally, %d of these failures did not provide error reports: %s".format(missingReports.size, missingReports mkString ",")
-              baseMessage + "\n" + supplementalErrorMessage
+      createDeleteResponse(Future sequence deleteFutures) pipeTo context.parent
+  }
+
+  def createDeleteResponse(deleteFutures: Future[Seq[HttpResponse]]): Future[PerRequestMessage] = {
+    deleteFutures.map
+      { responses =>
+        val allSucceeded = responses.forall(_.status.isSuccess)
+        allSucceeded match {
+          case true =>
+            RequestComplete(OK)
+          case false =>
+            val errors = responses.filterNot(_.status.isSuccess) map { e => (e, ErrorReport.tryUnmarshal(e)) }
+            val errorReports = errors collect { case (_, Success(report)) => report }
+            val missingReports = errors collect { case (originalError, Failure(_)) => originalError }
+            val errorMessage = {
+              val baseMessage = "%d entities deleted, with %d failures.  Errors: %s".format(entities.size, errors.size, errors mkString ",")
```

Comment:
Yeah, it bugs me too. I was hoping to avoid changing the code this much - the original code suffers from the same problem and I wanted to focus this task on just the actor refactoring. I'll make some tweaks across the board and squash.